Start c++ guidelines

lkeegan opened the pull request:
- c++ sensible default choices
- some 3rd party library recommendations
- github pages version: https://ssciwr.github.io/guidelines

lkeegan committed:
add cpp good default choices & 3rd party libs

lkeegan committed:
add jekyll config yml to set theme

lkeegan committed:
add cookiecutter-cpp paragraph & link

iulusoy commented:
I would also add a recommendation for documentation. I guess we agreed on doxygen -> sphinx via breathe at least for the coding part. We can recommend to publish to rtd or github pages for public repos. But it should be clear that just the doxygen output is not sufficient for a documentation. For python, the recommendation would directly be sphinx (as per the python template repo, the documentation is on rtd). For Fortran, it would also follow the same route as C++.

dokempf commented:
These recommendations all LGTM. Additional topics that come to my mind: Language standard recommendation (don't start a 03 project), Code Formatting. Once I finished the cookiecutter, we should think about how to present the template repo and the cookiecutter.

lkeegan committed:
add doxygen to default choices
  - as suggested by @iulusoy

add tooling section
  - with formatting as suggested by @dokempf
  - wth IDEs

lkeegan commented:
thanks for the feedback! @iulusoy I agree about docs being more than just e.g. doxygen, but I think this is not c++ specific but applies regardless of the language so it can go in the general guidelines
